projects/social/social.py

```diff
@@ -48,7 +51,22 @@ def populateGraph(self, numUsers, avgFriendships):
 
         # Add users
 
+        for i in range(numUsers):
+            self.addUser(str(i))
+
         # Create friendships
+        '''
+        To create N random friendships, you could create a list with all possible friendship combinations, shuffle the list, then grab the first N elements from the list
+        '''
+        friendship_list = []
+        for user in self.users:
+            for friend in range(user + 1, self.lastID + 1):
+                friendship_list.append((user, friend))
+        random.shuffle(friendship_list)
+
+        for i in range(0, (numUsers*avgFriendships//2)):
```

Comment:
Nice work with this problem, and understanding how many friendships to actually generate.  

Perhaps we could have more comments throughout this.  We don't need them for every line, but here and there to note what a for loop is doing or something else might be helpful in the future.